### ISSUE TITLE
Possibility to take result of urlparse/urlsplit as url

### ISSUE THREAD

**Issue by Yomguithereal** [barseghyanartur/tld#41]
It could be useful for performance reasons in some use cases.

What do you think @barseghyanartur?

**Comment by barseghyanartur** [comment]
Could you show the use-case? In general, sure, if that's helpful, why not? :)

**Comment by Yomguithereal** [comment]
Typically I work on web crawlers and when collecting links, I create hierarchical representations of urls so I can insert them into tries. To do so, I must parse them and split their elements so I can re-organize them in the correct order. So, I already use urlparse/urlsplit etc. to do the job. Thing is it would be nice if I could pass the parsed tuple to your functions so it does not need to re-parse the url once again.

**Comment by barseghyanartur** [comment]
@Yomguithereal:

I think we could add ``parsed_url`` as a property to the ``Result`` object. In case if someone needs it.

Look [here](https://github.com/barseghyanartur/tld/blob/master/src/tld/utils.py#L255).

We could do:

```python
parsed_url = urlsplit(url).netloc
domain_name = parsed_url.netloc
```

and then return parsed_url along with ``domain_parts`` and ``non_zero_i``.

``get_tld`` and ``get_fld`` shall change accordingly.

Finally, feed it further to the ``Result`` object as a ``parsed_url`` argument.

If you agree, could you make a pull request?

**Comment by barseghyanartur** [comment]
To be implemented in 0.9.

**Comment by barseghyanartur** [comment]
Released in 0.9.

**Comment by barseghyanartur** [comment]
```
    from tld import get_tld
    
    res = get_tld("http://some.subdomain.google.co.uk", as_object=True)

    res
    # 'co.uk'

    res.subdomain
    # 'some.subdomain'

    res.domain
    # 'google'

    res.tld
    # 'co.uk'

    res.fld
    # 'google.co.uk'

    res.parsed_url
    # SplitResult(
    #     scheme='http',
    #     netloc='some.subdomain.google.co.uk',
    #     path='',
    #     query='',
    #     fragment=''
    # )
```
